### PR TITLE
Add scale instancegroup command

### PR DESCRIPTION
**Pull request by gianrubio**
I did this PR to make easy to scale nodes as suggested on #1892. 

It introduce the `scale` command, so the user will be able to run `scale ig --name cluster.kops.ddy.systems nodes --replicas=4` to increase the instance size. 
It run faster than the rolling update because it just change the max/desired size of instance.

I'm open for discussions about!!

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2765)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by chrislovecnm** [comment]
You need to run `make gen-cli-docs` to create the documentation.

File: pkg/instancegroups/instancegroups.go [hunk 1]
```diff
@@ -446,3 +446,60 @@ func (g *CloudInstanceGroup) Delete(cloud fi.Cloud) error {
...
+			DesiredCapacity:      g.asg.DesiredCapacity,
+			MaxSize:              g.asg.MaxSize,
+		}
+		_, err := c.Autoscaling().UpdateAutoScalingGroup(request)
+
+		if err != nil {
+			return fmt.Errorf("error scaling autoscaling group %q: %v", asgName, err)
+		}
+
+		// TODO it's important to wait to this command apply or just print a message to the user is enough?
```
Comment (justinsb) [inline pkg/instancegroups/instancegroups.go]: I think printing a message is a great idea, rather than waiting.

**Comment by justinsb** [comment]
So I think this looks really good - can you `make gen-cli-docs` so we can merge it?  I _think_ that's all that is missing!

**Comment by gianrubio** [comment]
I add more validations also reviewed the docs. 

Thats the output:
```
$ ./.build/local/kops  scale ig nodes --replicas=0 && kops get ig nodes -o yaml
Using cluster from kubectl context: kops

I0929 08:25:49.677842   81183 instancegroups.go:493] Scaling InstanceGroup "nodes" from 5 to 0
I0929 08:25:49.677875   81183 instancegroups.go:499] Changing min instances to 0
I0929 08:25:49.844774   81183 scale_instancegroup.go:127] Successfull!
Using cluster from kubectl context: kops

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2017-08-02T13:46:39Z
  labels:
    kops.k8s.io/cluster: kops
  name: nodes
spec:
  image: 595879546273/CoreOS-stable-1465.8.0-hvm
  machineType: t2.xlarge
  maxSize: 5
  minSize: 0
  role: Node
  subnets:
  - eu-west-1a
  - eu-west-1b
giancarlo.rubio$

$ ./.build/local/kops  scale ig master-eu-west-1a --replicas=1 && kops get ig master-eu-west-1a -o yaml
Using cluster from kubectl context: kops

I0929 08:27:11.740500   81196 instancegroups.go:493] Scaling InstanceGroup "master-eu-west-1a" from 1 to 1
I0929 08:27:11.740527   81196 instancegroups.go:499] Changing min instances to 1
I0929 08:27:11.890719   81196 scale_instancegroup.go:127] Successfull!
Using cluster from kubectl context: kops

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2017-08-02T13:46:39Z
  labels:
    kops.k8s.io/cluster: kops
  name: master-eu-west-1a
spec:
  image: 595879546273/CoreOS-stable-1465.8.0-hvm
  machineType: t2.large
  maxSize: 1
  minSize: 1
  role: Master
  subnets:
  - eu-west-1a
```

File: cmd/kops/scale_instancegroup.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+type ScaleIgCmd struct {
+	Yes      bool
+	Replicas int64
+}
+
+const (
+	defReplicas = -1
+)
+
+// TODO add ability to add-nodes rather than scale
```
Comment (justinsb) [inline cmd/kops/scale_instancegroup.go]: Would this be like `kops scale ig nodes +2`?  That would indeed be cool (but let's get this PR merged!)

File: cmd/kops/scale_instancegroup.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	cmd := &cobra.Command{
+		Use:     "ig",
+		Aliases: []string{"instancegroup", "instancegroups"},
+		Short:   i18n.T("Scale instances instancegroups"),
+		Long:    scale_instancegroup_long,
+		Example: scale_instancegroup_example,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if len(args) == 0 {
+				exitWithError(fmt.Errorf("Specify name of instance group to edit"))
```
Comment (justinsb) [inline cmd/kops/scale_instancegroup.go]: Nit: probably should scale "to scale"

File: cmd/kops/scale_instancegroup.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	cluster, err := rootCommand.Cluster()
+	if err != nil {
+		return err
+	}
+
+	if groupName == "" {
+		return fmt.Errorf("name is required")
+	}
+
+	if c.Replicas == defReplicas {
```
Comment (justinsb) [inline cmd/kops/scale_instancegroup.go]: Nit: It's probably better to use the Changed field on the flag (doing this in the cobra.Command Run function instead) (https://github.com/spf13/cobra/issues/23).  The reason being that then the -1 default value won't appear in the help!

File: cmd/kops/scale_instancegroup.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+// TODO add ability to add-nodes rather than scale
+var (
+	//TODO add comments
+	scale_instancegroup_long = templates.LongDesc(i18n.T(`
+	long description...
+	`))
+
+	scale_instancegroup_example = templates.Examples(i18n.T(`
+		# Scale a ig fixing it to 2 replicas
+		kops scale ig --name cluster.kops.ddy.systems nodes --replicas=2
```
Comment (justinsb) [inline cmd/kops/scale_instancegroup.go]: Either $NAME or cluster.example.com

File: pkg/client/simple/clientset.go [hunk 6]
```diff
@@ -17,14 +17,15 @@ limitations under the License.
 package simple
 
 import (
+	"net/url"
+	"strings"
```
Comment (justinsb) [inline pkg/client/simple/clientset.go]: This is what triggered the rebase requirement!

File: pkg/instancegroups/instancegroups.go [hunk 7]
```diff
@@ -468,3 +468,77 @@ func (g *CloudInstanceGroup) Delete(cloud fi.Cloud) error {
...
+
+	if len(groups) != 1 {
+		return fmt.Errorf("Multiple InstanceGroup resources found in cloud")
+	}
+
+	cig.asg.DesiredCapacity = &c.DesiredReplicas
+	glog.Infof("Scaling InstanceGroup %q instances size to: %v", group.ObjectMeta.Name, c.DesiredReplicas)
+
+	// decrease max cluster size
+	if c.DesiredReplicas < *cig.asg.MinSize {
```
Comment (justinsb) [inline pkg/instancegroups/instancegroups.go]: Ah - tricky!  I'm thinking we should just set min and max both to the desired size.  If you disagree, can you add a comment (here or in the code) to explain?

File: pkg/instancegroups/instancegroups.go [hunk 8]
```diff
@@ -468,3 +468,77 @@ func (g *CloudInstanceGroup) Delete(cloud fi.Cloud) error {
...
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *CloudInstanceGroup) Scale(cloud fi.Cloud) error {
+
+	c := cloud.(awsup.AWSCloud)
```
Comment (justinsb) [inline pkg/instancegroups/instancegroups.go]: I worry we might have broken this with some of the refactorings of this interface that have gone in recently... it should be easier now, but let me know if you need some pointers figuring out where things have moved to!

**Comment by emmekappa** [comment]
any updates on this?

**Comment by justinsb** [comment]
I'm going to go ahead and close this PR.  If you do want to revive it @gianrubio it would be a great addition :-)

/close

**Pull request closed without merging**